src/org/zaproxy/zap/GuiBootstrap.java

```diff
@@ -345,15 +340,37 @@ private static void setDefaultViewLocale(Locale locale) {
      * {@link java.net.InetAddress InetAddress}) preventing some ZAP options from being correctly applied.
      */
     private void setupLookAndFeel() {
-        if (lookAndFeelSet) {
-            return;
+    	
+    	String lookAndFeelClassnameFromCommandline = System.getProperty("swing.defaultlaf");
+    	OptionsParam options = Model.getSingleton().getOptionsParam();
+    	String lookAndFeelnameFromOptions = options.getViewParam().getLookAndFeel();
+    	String lookAndFeelClassnameFromOptions = "";
+    	UIManager.LookAndFeelInfo[] looks = UIManager.getInstalledLookAndFeels();
+    	
+    	for(UIManager.LookAndFeelInfo index : looks) {
+    		if(index.getName().equals(lookAndFeelnameFromOptions)) {
+    			lookAndFeelClassnameFromOptions = index.getClassName();
+    			break;
+    		}
+    	}
+    	
+       if ( lookAndFeelClassnameFromCommandline!= null) {
+        	try{
+        		UIManager.setLookAndFeel(lookAndFeelClassnameFromCommandline);
+        		
+        		return;
+        	} catch (final UnsupportedLookAndFeelException
+        			| ClassNotFoundException
+        			| ClassCastException
+        			| InstantiationException
+        			| IllegalAccessException e) {
+        		logger.warn("Failed to set the specified look and feel:" + e.getMessage());
+        	}
         }
-        lookAndFeelSet = true;
-
-        String lookAndFeelClassname = System.getProperty("swing.defaultlaf");
-        if (lookAndFeelClassname != null) {
+       if (lookAndFeelClassnameFromOptions != null) {
```

Comment:
Should also check if not empty (that's the default).

Reply:
@thc202  Right now, if `lookAndFeelClassnameFromOptions` and `lookAndFeelClassnameFromCommandline` is empty , then code at https://github.com/zaproxy/zaproxy/blob/develop/src/org/zaproxy/zap/GuiBootstrap.java#L367,L388  sets the nimbus look. I will rewrite the entire the function.

Reply:
Yes, but no point trying to set `lookAndFeelClassnameFromOptions` if empty, it will fail, always (and a WARN is logged unnecessarily).